File: Data Structures & Algorithms/count-paths/submission-0.java
class Solution {

    HashMap<Integer, Integer> memo;

    public int uniquePaths(int m, int n) {
        memo = new HashMap<Integer, Integer>();

        return uniquePaths(m,n,0,0,0);
    }

    public int uniquePaths(int m, int n, int y, int x, int i) {
        if(y == m-1 && x == n-1) return 1;
        if(memo.get(y * n + x) != null) return memo.get(y * n + x);

        if(y < m-1) i += uniquePaths(m,n,y+1,x,0);
        if(x < n-1) i += uniquePaths(m,n,y,x+1,0);

        memo.put(y * n + x, i);

        return i;
    }
}

File: Data Structures & Algorithms/hand-of-straights/submission-0.py
class Solution:
    def isNStraightHand(self, hand: List[int], groupSize: int) -> bool:
        if len(hand) % groupSize > 0:
            return False

        if groupSize == 1:
            return True
        
        hand = sorted(hand)

        memo = {}

        for i in hand:
            if i not in memo.keys():
                memo[i] = 1
            else: 
                memo[i] += 1

        while memo:
            x = min(memo.keys());
            for i in range(x,x+groupSize):
                if i not in memo.keys():
                    return False
                memo[i] -= 1
                if memo[i] == 0:
                    memo.pop(i)
        
        return True

        

        